pkg/cloudprovider/nodecontroller/nodecontroller.go

```diff
@@ -253,6 +257,12 @@ func (nc *NodeController) Run(period time.Duration, syncNodeList bool) {
 			glog.Errorf("Error monitoring node status: %v", err)
 		}
 	}, nc.nodeMonitorPeriod)
+
+	go util.Forever(func() {
```

Comment:
In that case you might want to think about making this a more general queue (not in this pr, just as a consideration). Since I can see times when we realize a node is bad and want to take some offline action that isn't deletePods (eg: set status of pods to unknown).